sql/core/src/test/scala/org/apache/spark/sql/SubquerySuite.scala

```diff
@@ -1876,4 +1877,29 @@ class SubquerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
         "ReusedSubqueryExec should reuse an existing subquery")
     }
   }
+
+  test("SPARK-36280: Remove redundant aliases after RewritePredicateSubquery") {
+    sql("CREATE TABLE t1 USING parquet AS SELECT id AS a, id AS b, id AS c FROM range(10)")
+    sql("CREATE TABLE t2 USING parquet AS SELECT id AS x, id AS y FROM range(8)")
+    val df = sql(
+      """
+        |SELECT *
+        |FROM   t1
+        |WHERE  a IN (SELECT x
+        |             FROM   (SELECT x                         AS x,
+        |                            Rank()
+        |                              OVER (
+        |                                PARTITION BY x
+        |                                ORDER BY Sum(y) DESC) AS ranking
+        |                     FROM   t2
+        |                     GROUP  BY x) tmp1
+        |             WHERE  ranking <= 5)
```

Comment:
Do we use `https://www.dpriver.com/pp/sqlformat.htm` style in our other test cases? Is there a reason why you choose this formatter, @wangyum ?

Reply:
I have used this tool for many years. or do you have a recommended tool?

Reply:
I don't think we have a standard about the SQL format in Spark right now. This format looks fine, except for the `SELECT x  ...many spaces... AS x`